### PR TITLE
Remove legacy AtomicLong impl and IdGenerator

### PR DESCRIPTION
**Pull request by mdogan**
Additonally removed deprecated IdGenerator.

Moved old AtomicLong impl to an internal package
and renamed it to LongRegister. LongRegister still uses
the old service name to make it backward compatible
while doing perf testing.
LongRegister is using 255 (max id) as codec id.

See TDD: https://hazelcast.atlassian.net/wiki/spaces/EN/pages/1749057609/Concurrency+API+Clean-up+Design

EE: https://github.com/hazelcast/hazelcast-enterprise/pull/3174
Protocol change: https://github.com/hazelcast/hazelcast-client-protocol/pull/231

### CODE REVIEW

**Comment by mdogan** [comment]
run-lab-run

**Comment by pveentjer** [comment]
This PR is going to lead to very big problems to me. Because I will not be able to go to previous version of the 4.0 branch.

**Comment by pveentjer** [comment]
I don't care for the Id generator; fine to delete it. But moving of IAtomicLong will give me very big headache.

**Comment by mdogan** [comment]
@pveentjer: As we talked before, you will be able to use previous versions. You just need to change access to the `IAtomicLong`. Instead of `HazelcastInstace.getAtomicLong(name)`, you will use `HazelcastInstance.getDistributedObject("hz:impl:atomicLongService", name)`.

**Comment by pveentjer** [comment]
The package name of IAtomicLong will change.. right?

File: hazelcast/src/test/java/com/hazelcast/internal/longregister/LongRegisterAbstractTest.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    protected HazelcastInstance[] instances;
+    protected IAtomicLong longRegister;
+
+    @Before
+    public void setup() {
+        instances = newInstances();
+        HazelcastInstance local = instances[0];
+        HazelcastInstance target = instances[instances.length - 1];
+        String name = generateKeyOwnedBy(target);
+        longRegister = local.getDistributedObject(LongRegisterService.SERVICE_NAME, name);
```
Comment (mdogan) [inline hazelcast/src/test/java/com/hazelcast/internal/longregister/LongRegisterAbstractTest.java]: @pveentjer: see 👆

**Comment by mdogan** [comment]
Since https://github.com/hazelcast/hazelcast/pull/15143, `IAtomicLong` is in `com.hazelcast.cp` package, it will still remain there.

File: hazelcast/src/main/java/com/hazelcast/internal/longregister/client/codec/LongRegister.yaml [hunk 2]
```diff
@@ -0,0 +1,212 @@
+id: 255
```
Comment (ihsandemir) [inline hazelcast/src/main/java/com/hazelcast/internal/longregister/client/codec/LongRegister.yaml]: Should we keep this file at protocol repo as internal hidden?

Comment (mdogan) [reply hazelcast/src/main/java/com/hazelcast/internal/longregister/client/codec/LongRegister.yaml]: @asimarslan requested me to remove it from the official protocol repo. So I decided to keep it here.

Comment (asimarslan) [reply hazelcast/src/main/java/com/hazelcast/internal/longregister/client/codec/LongRegister.yaml]: I think we won't need this file anymore. The legacy `atomicLong` codecs shouldn't be changed and even if they are changed we don't support any kind of backward compatibility.

Comment (mdogan) [reply hazelcast/src/main/java/com/hazelcast/internal/longregister/client/codec/LongRegister.yaml]: I know it's not needed but still I want to keep it as a documentation of codecs.

**Pull request merged**